src/advanced_agent/memory/memory_cleaner.py

```
@@ -0,0 +1,382 @@
+            for i, memory1 in enumerate(memories):
+                if memory1.get('id') in processed_ids:
+                    continue
+                
+                content1 = memory1.get('content', '')
+                if len(content1) < self.cleanup_rules["min_content_length"]:
+                    continue
+                
+                embedding1 = self.embedding_manager.encode_text(content1)
+                duplicate_group = [memory1.get('id')]
+                
+                for j, memory2 in enumerate(memories[i+1:], i+1):
```

Ruff (0.12.2)

Loop control variable `j` not used within loop body

Rename unused `j` to `_j`

(B007)